tgui/packages/tgui/interfaces/ZoldorfPlayerShop/index.tsx

```diff
@@ -0,0 +1,128 @@
+                          disabled={product.soul_percentage > data.user_soul}
+                          onClick={() => act('soul_purchase', { "item": product.name })}
+                          style={{ "width": "50px", "text-align": "center", "padding": "0px" }}
+                          content={("" + product.soul_percentage + "%")}
+                        />
+                      </ZoldorfProductList>
+                    );
+                  })
+                }
+              </Stack>
+              <Stack vertical>
+                {
+                  credit_products.map(product => {
+                    return (
+                      <ZoldorfProductList {...product} key={product.name}>
+                        <Button
+                          color="green"
+                          disabled={product.price > credits}
+                          content={("" + product.price + "⪽")}
+                          onClick={() => act('credit_purchase', { "item": product.name })}
+                          style={{ "width": "50px", "text-align": "center", "padding": "0px" }}
+                        />
+                      </ZoldorfProductList>
+                    );
+                  })
+                }
+              </Stack>
+            </Section>
+          </Stack.Item>
+          { credits !== 0 && (
+            <Stack.Item>
+              <Table>
+                <Table.Row>
+                  <Table.Cell bold direction="row">
+                    {("Cash: " + credits + "⪽")}
+                    <Button
+                      icon="eject"
+                      ml="1%"
+                      content={"eject"}
+                      onClick={() => act('returncash')}
+                    />
+                  </Table.Cell>
+                </Table.Row>
+              </Table>
+            </Stack.Item>
+          )}
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const ZoldorfProductList = (props: ZoldorfProductListProps) => {
+  const {
+    name,
+    img,
+    stock,
+    infinite,
+    children,
+  } = props;
+  return (
+    <Flex justify="space-between" align="stretch" style={{ "border-bottom": "1px #555 solid" }}>
+      <Flex.Item direction="row">
+        {img && (
+          <Box style={{ "overflow": "show", "height": "24px" }}>
+            <img
+              src={`data:image/png;base64,${img}`}
+              style={{
+                'transform': 'translate(0, -4px)',
+              }}
+            />
+          </Box>)}
+      </Flex.Item>
+      <Flex.Item direction="row"
+        grow style={{
+          "display": "flex",
+          "justify-content": "center",
+          "flex-direction": "column",
+        }}>
+        <Box>
+          <Box inline italic>
+            {!infinite && `${stock} x`}&nbsp;
+          </Box>
+          <Box inline>{name}</Box>
+        </Box>
+      </Flex.Item>
+      <Flex.Item bold direction="row" style={{ "margin-left": "5px",
```

Comment:
Why do you need to change the `margin-left` here? i.e. what is wrong with base behavior? I'm not sure if `direction="row"` does anything here (it might do by virtue of being one of the properties that gets straight converted to a CSS property), but in either case is at odds with the `flex-direction: column;` set further down.

Is the goal is to center the items within each flex cell?

You probably want `align="center"` (for vertical centering) on the containing `Flex` at least, that should remove some of these style overrides.